src/registrar/models/contact.py

```diff
@@ -54,6 +54,50 @@ class Contact(TimeStampedModel):
         db_index=True,
     )
 
+    def has_more_than_one_join(self, expected_relation):
+        """Helper for finding whether an object is joined more than once.
+        expected_relation is the one relation with one expected join"""
+        # all_relations is the list of all_relations (from contact) to be checked for existing joins
+        all_relations = [
+            "user",
+            "authorizing_official",
+            "submitted_applications",
+            "contact_applications",
+            "information_authorizing_official",
+            "submitted_applications_information",
+            "contact_applications_information",
+        ]
+        return any(self._has_more_than_one_join_per_relation(rel, expected_relation) for rel in all_relations)
+
+    def _has_more_than_one_join_per_relation(self, relation, expected_relation):
+        """Helper for finding whether an object is joined more than once."""
+        # threshold is the number of related objects that are acceptable
+        # when determining if related objects exist. threshold is 0 for most
+        # relationships. if the relationship is expected_relation, we know that
+        # there is already exactly 1 acceptable relationship (the one we are
+        # attempting to delete), so the threshold is 1
+        threshold = 1 if relation == expected_relation else 0
+
+        # Raise a KeyError if rel is not a defined field on the db_obj model
+        # This will help catch any errors in relation passed.
+        if relation not in [field.name for field in self._meta.get_fields()]:
+            raise KeyError(f"{relation} is not a defined field on the {self._meta.model_name} model.")
+
+        # if attr rel in db_obj is not None, then test if reference object(s) exist
+        if getattr(self, relation) is not None:
+            field = self._meta.get_field(relation)
+            if isinstance(field, models.OneToOneField):
+                # if the rel field is a OneToOne field, then we have already
+                # determined that the object exists (is not None)
+                # so return True unless the relation being tested is the expected_relation
+                return True if relation != expected_relation else False
```

Comment:
```suggestion
                is_not_expected_relation = relation != expected_relation
                return is_not_expected_relation 
```
(Q) is this returning `return True if relation != expected_relation else False` for code legibility purposes? If so, I would recommend returning a boolean with a descriptive variable name (as the example above shows)

Reply:
Good change